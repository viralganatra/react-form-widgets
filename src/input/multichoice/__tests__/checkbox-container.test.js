import React from 'react';
import CheckboxContainer from '../checkbox-container';
import GroupComponent from '../__mocks__/component-group';

jest.mock('../../../events');

const data = {
    items: [
        { label: 'test 1', value: '1' },
        { label: 'test 2', value: '2' },
    ],
    value: ['2'],
};

describe('CheckboxContainer', () => {
    let Composed;

    beforeEach(() => {
        Composed = CheckboxContainer(GroupComponent);
    });

    it('should render correctly', () => {
        const wrapper = shallow(<Composed {...data} />);

        expect(() => CheckboxContainer({})).toThrowErrorMatchingSnapshot();
        expect(wrapper).toMatchSnapshot();
    });

    it('should validate prop types', () => {
        const error = console.error;
        console.error = jest.fn();

        shallow(<Composed />);
        shallow(<Composed type="text" />);
        shallow(<Composed value={2} />);

        expect(console.error.mock.calls[0][0]).toBeDefined();
        expect(console.error.mock.calls[1][0]).toBeDefined();
        expect(console.error.mock.calls[2][0]).toBeDefined();

        console.error = error;
    });

    it('should listen to the onChange event and return an array of checked item values', () => {
        const methodSpy = jest.fn();
        const wrapper = mount(<Composed {...data} onChange={methodSpy} />);

        wrapper.find('input').at(0).simulate('change', { target: { value: '1' } });

        expect(wrapper).toMatchSnapshot();

        wrapper.find('input').at(0).simulate('change', { target: { value: '1' } });
        wrapper.find('input').at(1).simulate('change', { target: { value: '2' } });

        expect(methodSpy.mock.calls[0][0].value).toEqual(['2', '1']);
        expect(methodSpy.mock.calls[1][0].value).toEqual(['2']);
        expect(methodSpy.mock.calls[2][0].value).toEqual([]);

        expect(wrapper).toMatchSnapshot();
    });
});
