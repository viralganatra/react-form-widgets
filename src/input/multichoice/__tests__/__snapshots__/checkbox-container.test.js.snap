exports[`CheckboxContainer should listen to the onChange event and return an array of checked item values 1`] = `
<CheckboxContainer
  items={
    Array [
      Object {
        "label": "test 1",
        "value": "1",
      },
      Object {
        "label": "test 2",
        "value": "2",
      },
    ]
  }
  onBlur={[Function]}
  onChange={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyPress={[Function]}
  onKeyUp={[Function]}
  type="checkbox"
  value={
    Array [
      "2",
    ]
  }>
  <GroupComponent
    isChecked={[Function]}
    items={
      Array [
        Object {
          "label": "test 1",
          "value": "1",
        },
        Object {
          "label": "test 2",
          "value": "2",
        },
      ]
    }
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyPress={[Function]}
    onKeyUp={[Function]}
    type="checkbox">
    <div>
      <input
        checked={true}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyPress={[Function]}
        onKeyUp={[Function]}
        type="checkbox"
        value="1" />
      <input
        checked={true}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyPress={[Function]}
        onKeyUp={[Function]}
        type="checkbox"
        value="2" />
    </div>
  </GroupComponent>
</CheckboxContainer>
`;

exports[`CheckboxContainer should listen to the onChange event and return an array of checked item values 2`] = `
<CheckboxContainer
  items={
    Array [
      Object {
        "label": "test 1",
        "value": "1",
      },
      Object {
        "label": "test 2",
        "value": "2",
      },
    ]
  }
  onBlur={[Function]}
  onChange={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyPress={[Function]}
  onKeyUp={[Function]}
  type="checkbox"
  value={
    Array [
      "2",
    ]
  }>
  <GroupComponent
    isChecked={[Function]}
    items={
      Array [
        Object {
          "label": "test 1",
          "value": "1",
        },
        Object {
          "label": "test 2",
          "value": "2",
        },
      ]
    }
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyPress={[Function]}
    onKeyUp={[Function]}
    type="checkbox">
    <div>
      <input
        checked={false}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyPress={[Function]}
        onKeyUp={[Function]}
        type="checkbox"
        value="1" />
      <input
        checked={false}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyPress={[Function]}
        onKeyUp={[Function]}
        type="checkbox"
        value="2" />
    </div>
  </GroupComponent>
</CheckboxContainer>
`;

exports[`CheckboxContainer should render correctly, validating prop types 1`] = `
"Expected \"WrappedComponent\" provided as the first argument to
        CheckboxContainerHOC to be a function. Instead, received
        object."
`;

exports[`CheckboxContainer should render correctly, validating prop types 2`] = `
<GroupComponent
  isChecked={[Function]}
  items={
    Array [
      Object {
        "label": "test 1",
        "value": "1",
      },
      Object {
        "label": "test 2",
        "value": "2",
      },
    ]
  }
  onBlur={[Function]}
  onChange={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyPress={[Function]}
  onKeyUp={[Function]}
  type="checkbox" />
`;
